custom_components/openmediavault/binary_sensor.py

```diff
@@ -82,54 +82,26 @@ def __init__(self, omv_controller, inst, sensor):
 
         self._state = None
         self._icon = None
-        self._attrs = {ATTR_ATTRIBUTION: ATTRIBUTION}
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return f"{self._inst} {self._type[ATTR_LABEL]}"
-
-    @property
-    def device_state_attributes(self):
-        """Return the state attributes."""
-        return self._attrs
-
-    @property
-    def unique_id(self):
-        """Return a unique_id for this entity."""
-        return f"{self._inst.lower()}-{self._sensor.lower()}"
-
-    @property
-    def available(self) -> bool:
-        """Return if controller is available."""
-        return self._ctrl.connected()
-
-    @property
-    def device_info(self):
-        """Return a description for device registry."""
-        info = {
+        self._attr_extra_state_attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
+        self._attr_name = f"{self._inst} {self._type[ATTR_LABEL]}"
+        self._attr_unique_id = f"{self._inst.lower()}-{self._sensor.lower()}"
+        self._attr_available = self._ctrl.connected()
+        self._attr_is_on = False
+        if self._attr in self._data:
+            self._attr_is_on = self._data[self._attr]
+
+        self._attr_device_info = {
             "manufacturer": "OpenMediaVault",
             "name": f"{self._inst} {self._type[ATTR_GROUP]}",
         }
         if ATTR_GROUP in self._type:
-            info["identifiers"] = {
+            self._attr_device_info["identifiers"] = {
                 (DOMAIN, self._inst, "sensor", self._type[ATTR_GROUP])
             }
 
-        return info
-
     async def async_update(self):
         """Synchronize state with controller."""
 
     async def async_added_to_hass(self):
         """Entity created."""
         _LOGGER.debug("New sensor %s (%s)", self._inst, self._sensor)
-
-    @property
-    def is_on(self):
```

Comment:
this would break default mass binary sensors. I dont think they are used in OMV at the moment, but they may be later.

Reply:
I wasn't sure about this one, will revert

Reply:
fixed